appveyor.yml

```diff
@@ -0,0 +1,166 @@
+        mv .\libjpeg*\*.h $DEP_DIR\include\
+        mv .\libjpeg*\*.lib $DEP_DIR\lib\
+        mv .\libjpeg*\*.dll $DEP_DIR\lib\
+
+        nuget install freetype
+        mv .\freetype.2.8.0.1\build\native\include\* $DEP_DIR\include\
+        mv .\freetype.2.8.0.1\build\native\lib\$env:PLATFORM\v140\dynamic\Release\*.lib $DEP_DIR\lib\
+        mv .\freetype.redist.2.8.0.1\build\native\bin\$env:PLATFORM\v140\dynamic\Release\*.lib $DEP_DIR\lib\
+
+        # ZLib
+        cd $DWN_DIR
+        Start-FileDownload https://github.com/madler/zlib/archive/v1.2.8.tar.gz
+        tar -xzf v1.2.8.tar.gz; cd zlib-1.2.8; md "$OSL_INT_DIR" -Force | Out-Null; cd "$OSL_INT_DIR"
+        cmake ..\.. -G "$GENERATOR" -DCMAKE_CONFIGURATION_TYPES="$env:CONFIGURATION" -DCMAKE_PREFIX_PATH="$DEP_DIR" -DCMAKE_INSTALL_PREFIX="$DEP_DIR"
+        cmake --build . --config $env:CONFIGURATION --target INSTALL
+
+        # IlmBase
+        cd $DWN_DIR
+        Start-FileDownload http://download.savannah.nongnu.org/releases/openexr/ilmbase-2.2.1.tar.gz
+        tar -xzf ilmbase-2.2.1.tar.gz ; cd ilmbase* ; md "$OSL_INT_DIR" -Force | Out-Null; cd "$OSL_INT_DIR"
+        cmake ..\.. -G "$GENERATOR" -DCMAKE_CONFIGURATION_TYPES="$env:CONFIGURATION" -DCMAKE_PREFIX_PATH="$DEP_DIR" -DCMAKE_INSTALL_PREFIX="$DEP_DIR"
+        cmake --build . --config $env:CONFIGURATION --target INSTALL
+
+        # OpenEXR
+        cd $DWN_DIR
+        Start-FileDownload http://download.savannah.nongnu.org/releases/openexr/openexr-2.2.1.tar.gz
+        tar -xzf openexr-2.2.1.tar.gz ; cd openexr* ; md "$OSL_INT_DIR" -Force | Out-Null; cd "$OSL_INT_DIR"
+        cmake ..\.. -G "$GENERATOR" -DCMAKE_CONFIGURATION_TYPES="$env:CONFIGURATION" -DCMAKE_PREFIX_PATH="$DEP_DIR" -DCMAKE_INSTALL_PREFIX="$DEP_DIR" -DILMBASE_PACKAGE_PREFIX="$DEP_DIR"
+        cmake --build . --config $env:CONFIGURATION --target INSTALL
+
+        # OpenImageIO
+        cd $DWN_DIR
+        Start-FileDownload https://github.com/OpenImageIO/oiio/archive/Release-1.8.7.zip
+        unzip Release-1.8.7.zip; cd .\oiio-Release*\; md "$OSL_INT_DIR" -Force | Out-Null; cd "$OSL_INT_DIR"
+        cmake ..\.. -G "$GENERATOR" -DCMAKE_CONFIGURATION_TYPES="$env:CONFIGURATION" -DCMAKE_PREFIX_PATH="$DEP_DIR;$BOOST_ROOT" -DCMAKE_INSTALL_PREFIX="$DEP_DIR" -DFREETYPE_DIR="$DEP_DIR" -DBOOST_LIBRARYDIR="$BOOST_LIBRARYDIR" -DUSE_PYTHON=OFF -DOIIO_BUILD_TESTS=OFF -DUSE_NUKE=OFF
+        cmake --build . --config $env:CONFIGURATION --target INSTALL
+
+        # LLVM
+        cd $DWN_DIR
+        git clone --depth 10 --branch release_40 https://github.com/llvm-mirror/llvm.git
+        cd llvm; md "$OSL_INT_DIR" -Force | Out-Null; cd "$OSL_INT_DIR"
+        cmake ..\.. -G "$GENERATOR" -DCMAKE_CONFIGURATION_TYPES="$env:CONFIGURATION" -DCMAKE_PREFIX_PATH="$DEP_DIR" -DCMAKE_INSTALL_PREFIX="$DEP_DIR" -DLLVM_TARGETS_TO_BUILD="X86"
```

Comment:
I'm wondering why you chose to build LLVM from source, which seems to me like it would end up being the majority of time of the CI build and test, possibly pushing it over the time limit and terminating the job early.

Would it not be better to grab pre-built binaries from here?
  http://releases.llvm.org/download.html